app/services/notifications/update.rb

```diff
@@ -23,7 +23,7 @@ def call
       # to load all of them in memory with `.blank?`, thus we choose `.none?`
       return if notifications.none?
 
-      new_json_data = notifications.first.json_data || {}
+      new_json_data = {}
```

Comment:
Further down on line 32, all of the notifications are updated with the new `json_data`.
https://github.com/thepracticaldev/dev.to/blob/ef3ab8b80966bdbdc0c9e8020033c6d985d73ecc/app/services/notifications/update.rb#L32
It seems like we are overwriting any existing `json_data` with the latest, so we don't need to query for any existing `json_data`.

Reply:
It's entirely possible that there is some context I don't have here about a key that could exist in `json_data` that we want to retain. Poking around the codebase, I didn't see where in the context of notifications on `Article` and `Comment` where that was true. It seems the only data that exists in this `json_data` column is the same as what we're updating it with.

Reply:
I tagged @lightalloy as a reviewer, I think she probably know the context for this, but to me it makes a lot of sense. I thought we were fetching the first notification because we were merging pre-existing data with new data

Reply:
I looked back at my changes https://github.com/thepracticaldev/dev.to/commit/dd0fbf002b0c151ef12c1422ce0aa808631579ca#diff-51859d5049905fb15f009d82b0b680cb and found out that I wasn't changing the logic then, just moved it. 
Potentially, yes, we could be merging data, but it seems like we are overwriting all the possible fields of `json_data` in the following lines anyway: https://github.com/thepracticaldev/dev.to/blob/4ca5932dc0908a2c83c736f1cd3913e612e3888a/app/services/notifications/update.rb#L27-L30

Reply:
@lightalloy that is the conclusion we came to as well in [our very long slack thread](https://dev-internal.slack.com/archives/CPGJMAX5F/p1581023014037700)